src/BaGet.Core/Configuration/MirrorOptions.cs

```diff
@@ -17,6 +17,11 @@ public class MirrorOptions : IValidatableObject
         /// </summary>
         public Uri PackageSource { get; set; }
```

Comment:
Hey this is a tiny bit of feature creep, but I think you can leverage the options validation here by adding attributes on the `PackageSource` property:

```diff
/// <summary>
/// The v3 index that will be mirrored.
/// </summary>
[RequiredIf(nameof(Enabled))]
[RequiredIf(nameof(Legacy))]
public Uri PackageSource { get; set; }

Reply:
Excellent idea. However when trying this it complains that only one `IfRequired` attribute can be used.

As I workaround, we can do the following, that will work but will show the error message `Invalid 'Mirror' configs: 'PackageSource' is required because 'EnabledOrLegacy' has a value of 'True'.` which does not reflect the settings properly, but maybe this is OK?

```
/// <summary>
/// If true, packages that aren't found locally will be indexed
/// using the upstream source.
/// </summary>
public bool Enabled { get; set; }

/// <summary>
/// The v3 index that will be mirrored.
/// </summary>
[RequiredIf(nameof(EnabledOrLegacy), true)]
public Uri PackageSource { get; set; }

/// <summary>
/// Whether or not the package source is a v2 package source feed.
/// </summary>
public bool Legacy { get; set; }

[NotMapped]
public bool EnabledOrLegacy => Enabled || Legacy;
```

Reply:
Ah good catch! It looks like you can support multiple attribute usages if we:

1. Set the attribute's `AttributeUsageAttribute.AllowMultiple` to true
1. Override the `TypeId` property (see [this](https://stackoverflow.com/questions/553540/how-to-create-duplicate-allowed-attributes))

For an example of a validation attribute that supports multiple uses, see [`CustomValidationAttribute`](https://github.com/dotnet/runtime/blob/b51270d64f833b87d33639341392eb8f35442efa/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/CustomValidationAttribute.cs). I'll do this in a follow-up, I've blocked your changes for a little too long already!